src/Courses/TopicGrades/ExportAllButton.tsx

```diff
@@ -83,6 +84,33 @@ export const ExportAllButton: React.FC<ExportAllButtonProps> = ({topicId, userId
         }
     };
 
+    const allOnClickActions = () => {
+        switch(buttonState) {
+        case ButtonOptions.DOWNLOAD:
+            if (!url) {
+                logger.warn('Download option should not appear if no url is present.');
+                return;
+            }
+            window.open(`/${url}`, '_blank');
+            break;
+        case ButtonOptions.PRINT_SINGLE:
+        case ButtonOptions.PRINT_SINGLE_NO_SOLUTIONS:
+            if (!userId) {
+                logger.warn('Print Single option should not appear if no userId is selected.');
+                return;
+            }
+            
+            window.open(`${path}/print/${userId}?${qs.stringify({showSolutions: buttonState === ButtonOptions.PRINT_SINGLE})}`, '_blank');
+            break;
+        case ButtonOptions.PRINT_BLANK:
+        case ButtonOptions.PRINT_BLANK_NO_SOLUTIONS:
+            window.open(`${path}/print/?${qs.stringify({showSolutions: buttonState === ButtonOptions.PRINT_BLANK})}`, '_blank');
+            break;
+        default:
+            checkAndStatusUpdateExport(true);
```

Comment:
I think the default should be reserved for error handling but here it looks like it is being used to account for
It looks like this is being used for export all?

If someone adds a key i can see this resulting in bugs